conans/test/functional/toolchains/microsoft/test_msbuild.py

```diff
@@ -408,7 +425,22 @@ def _run_app(client, arch, build_type, shared=None):
                            ("msvc", "19.0", "static", "14")]
                           )
     @pytest.mark.tool_cmake
-    def test_toolchain_win(self, compiler, version, runtime, cppstd):
+    def test_toolchain_win_vs2017(self, compiler, version, runtime, cppstd):
+        if self.vs_version != "15":
+            pytest.skip("test for Visual Studio 2017")
```

Comment:
Better put the skip as ``@pytest.mark``, even if the ``@parameterized_class(vs_versions)`` is removed and a bit of duplication of the parameterized happens for the tests.

Reply:
I was trying to make it work different ways, starting from copying the whole file, realizing it will be a nightmare to support, then tried some thing with `parameterized`. the main problem here I can't write:
```
@parameterized_class(vs_versions)
@pytest.mark.tool_visual_studio(version=vs_version)
class MSBuildGeneratorTest(unittest.TestCase):
```
or
```
    @parameterized.expand(self.vs_combo)
    def test_toolchain_win(self, compiler, version, runtime, cppstd):
```
in other words, I don't see how can I access `self` or `cls` from the decorator. it seems like decorator should be evaluated on the earlier state, thus it cannot reference class internals. I don't want to dig into this right now too much, it's already taking time, but current state works and provides tests for both vs2017 and vs2022 (as it was requested), also makes it relatively easy to add new version on demand.
so, `@pytest.mark` (or other decorator) cannot reference the version it needs to skip from the class, so it's much easier for now to skip inside the function without fighting with python internals.